File: src/main/java/org/zkoss/zkforge/aggrid/AggridDefaultColumn.java
/* AggridDefaultColumn.java

	Purpose:
		
	Description:
		
	History:
		Tue Sep 29 16:15:51 CST 2020, Created by rudyhuang

Copyright (C) 2020 Potix Corporation. All Rights Reserved.
*/
package org.zkoss.zkforge.aggrid;

import java.util.Comparator;

import org.zkoss.zkforge.aggrid.filter.ColumnFilters;

/**
 * @author rudyhuang
 */
public class AggridDefaultColumn extends Aggridcolumn<Object> {
	@Override
	public void setSortAscending(Comparator<Object> sorter) {
		throw new UnsupportedOperationException("not supported in default column definition");
	}

	@Override
	public void setSortDescending(Comparator<Object> sorter) {
		throw new UnsupportedOperationException("not supported in default column definition");
	}

	@Override
	protected void initSortComparator() {
		// do nothing
	}

	@Override
	protected void initFilterParams() {
		ColumnFilters.putFilterParams(AggridDefaultColumn.class.getName(), getFilterParams());
	}

	@Override
	protected boolean isChildable() {
		return false;
	}
}

File: src/main/resources/web/js/aggrid/Aggriddefaultcolumn.ts
/* Aggriddefaultcolumn.ts

	Purpose:
		
	Description:
		
	History:
		Tue Sep 29 16:28:36 CST 2020, Created by rudyhuang

Copyright (C) 2020 Potix Corporation. All Rights Reserved.
*/
// @ts-ignore
import {ColDef} from '@ag-grid-community/core';

(() => {
aggrid.Aggriddefaultcolumn = zk.$extends(aggrid.Aggridcolumn, {
	toColDef(): ColDef {
		return zk.copy({}, this._colDef);
	}
});
})();